subworkflows/local/metagenomics_profiling.nf

```diff
@@ -0,0 +1,174 @@
+//
+// Complexity filtering and metagenomics screening of sequencing reads
+//
+
+// Much taken from nf-core/taxprofile subworkflows/local/profiling.nf
+
+include { MALT_RUN                       } from '../../modules/nf-core/malt/run/main'
+include { KRAKEN2_KRAKEN2                } from '../../modules/nf-core/kraken2/kraken2/main'
+include { KRAKENUNIQ_PRELOADEDKRAKENUNIQ } from '../../modules/nf-core/krakenuniq/preloadedkrakenuniq/main'
+include { METAPHLAN_METAPHLAN            } from '../../modules/nf-core/metaphlan/metaphlan/main'
+include { CAT_CAT as CAT_CAT_MALT        } from '../../modules/nf-core/cat/cat/main'
+
+workflow METAGENOMICS_PROFILING {
+
+    take:
+    ch_reads
+    ch_database
+
+    main:
+    ch_versions             = Channel.empty()
+    ch_multiqc_files        = Channel.empty()
+    ch_postprocessing_input = Channel.empty()
+
+    /*
+        PREPARE PROFILER INPUT CHANNELS & RUN PROFILING
+    */
+
+    // Each tool has a slightly different input structure and generally separate
+    // input channels for reads vs database. We restructure the channel tuple
+    // for each tool and make liberal use of multiMap to keep reads/database
+    // channel element order in sync with each other
+
+    if ( params.metagenomics_profiling_tool == 'malt' ) {
+
+        // Optional parallel run of malt available:
+        // If parallel execution, split into groups with meta id of the first library id of group
+        // Merging of maltlog will be done by concatenation
+
+        // If no parallel execution (default):
+        // Reset entire input meta for MALT to just database name,
+        // as we don't run run on a per-sample basis due to huge databases
+        // so all samples are in one run and so sample-specific metadata
+        // unnecessary. Set as database name to prevent `null` job ID and prefix.
+
+
+        def label = file(params.metagenomics_profiling_database).getBaseName()
+
+        // For the next step we need the number of groups for the spezified number of input files
+        ch_groups =  params.metagenomics_malt_group_size > 0 ? ch_reads.collate(params.metagenomics_malt_group_size).count() : Channel.of(1)
+        // this is for enumerating the channel-entries in the ch_reads channel
+        def n = 0
```

Comment:
Please use a more descriptive variable name

Reply:
changed to groups_counter